app/integrations/aws/dynamodb.py

```diff
@@ -14,29 +14,95 @@
 
 @handle_aws_api_errors
 def query(
-    table_name,
-    key_condition_expression=None,
-    expression_attribute_values=None,
+    TableName,
     **kwargs,
 ):
-    params = {"TableName": table_name, "config": client_config}
-    if key_condition_expression:
-        params["KeyConditionExpression"] = key_condition_expression
-    if expression_attribute_values:
-        params["ExpressionAttributeValues"] = expression_attribute_values
+    params = {
+        "TableName": TableName,
+    }
+    if params:
+        params.update(kwargs)
+    response = execute_aws_api_call(
+        "dynamodb", "query", paginated=True, client_config=client_config, **params
+    )
+    return response
+
+
+@handle_aws_api_errors
+def scan(TableName, **kwargs):
+    params = {
+        "TableName": TableName,
+    }
     if kwargs:
         params.update(kwargs)
-    response = execute_aws_api_call("dynamodb", "query", **params)
+    response = execute_aws_api_call(
+        "dynamodb",
+        "scan",
+        paginated=True,
+        keys=["Items"],
+        client_config=client_config,
+        **params,
+    )
     return response
 
 
 @handle_aws_api_errors
-def scan(table, **kwargs):
+def put_item(TableName, **kwargs):
```

Comment:
I really like the rename of the function since "scan" was more ambiguous!

Reply:
I've simply hoisted the commands from the client as they are very context dependant. this way we can just invoke them directly but you can always go back to using the whole client if you prefer.